src/core/thread/network_data_service.cpp

```diff
@@ -302,28 +335,21 @@ Error Manager::GetNextDnsSrpUnicastInfo(Iterator &aIterator, DnsSrpUnicastType a
             // (then we parse and return the info), or it can be
             // empty (then we skip over it).
 
-            aIterator.mServerSubTlv->GetServerData(data);
-
-            if (data.GetLength() >= sizeof(DnsSrpUnicast::ServerData))
+            if (DnsSrpUnicast::ServerData::ParseFrom(*aIterator.mServerSubTlv, aInfo) == kErrorNone)
             {
-                const DnsSrpUnicast::ServerData *serverData =
-                    reinterpret_cast<const DnsSrpUnicast::ServerData *>(data.GetBytes());
-
-                aInfo.mSockAddr.SetAddress(serverData->GetAddress());
-                aInfo.mSockAddr.SetPort(serverData->GetPort());
-                aInfo.mRloc16 = aIterator.mServerSubTlv->GetServer16();
                 ExitNow();
             }
 
-            if (data.GetLength() == sizeof(uint16_t))
+            if (aIterator.mServerSubTlv->GetServerDataLength() == sizeof(uint16_t))
```

Comment:
This case is not specified (I think) - so why would it be useful to keep the code? It can only be used then in an 100% OpenThread network i.e. custom installation. BRs normally wouldn't publish this format. (Or did I miss something?)

Reply:
Yes. This is left for backward compatibility. This approach was used in OT early on (before it was clarified in the specification). We may want to remove it at some point, but I prefer to do this in its own PR, so I kept it in the code for now.